### PR TITLE
move from travis to github actions

### PR DESCRIPTION
**Pull request by paulirish**
move from travis to github actions.

this is on top of #27. that diff:  https://github.com/paulirish/git3po/compare/dropbabel...paulirish:gha

you can see the GHA action results here: https://github.com/paulirish/git3po/actions?query=branch%3Agha

i'm not 100% sure about the publish workflow being what you intend, @patrickhulce but it seems about right…

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: publish
+
+on:
+  push:
+    branches: [master]
```
Comment (patrickhulce) [inline .github/workflows/publish.yml:5]: this works, I typically share with CI and just add an if to the `npm run semantic-release` step, but I don't feel strongly 🤷

**Review by patrickhulce — COMMENTED** [review]
sweet, thanks!

File: .github/workflows/publish.yml (hunk 1 above)
Comment (paulirish) [reply .github/workflows/publish.yml:5]: yeah i saw that in the travis.yml.... 

if we kept it in the same workflow, wouldn't it be trying to publish from PR branches?  i feel like there's some distinction between travis & gha with push/pr stuff that matters here. 


oh oh you mean an `if branch == master` conditional on that semantic-release item.  didn't know that's a thing! but if so, then yeah that feels better.

**Comment by paulirish** [comment]
btw updated the PR title... it was very much innaccurate before. :p

File: .github/workflows/publish.yml (hunk 1 above)
Comment (patrickhulce) [reply .github/workflows/publish.yml:5]: > oh oh you mean an if branch == master conditional on that semantic-release item. didn't know that's a thing! but if so, then yeah that feels better.

Yep! 👍

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    - run: npm run test:lint
+    - run: npm run test:coverage
+
+    - run: nyc report --reporter=text-lcov | coveralls || echo 'Failed to upload to coveralls...'
+
+  release:
+    runs-on: ubuntu-latest
+    name: release
+    needs: [basics]
```
Comment (paulirish) [inline .github/workflows/ci.yml:35]: i haven't really validated this works, but... this dude uses this style: https://github.com/alekspog/cy-github-actions/blob/460b4b62761295b151c6453685b64b16dff53906/.github/workflows/main.yml#L33-L36

seems cool.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    # only release from the master branch
+    if: github.ref == 'refs/heads/master'
+    steps:
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 16.x
+    - run: npm ci
+    - run: npm run semantic-release
```
Comment (paulirish) [inline .github/workflows/ci.yml:43]: i feel like we probably need some 

```
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

here? shrug. ive never worked with this semantic-release thang

Comment (patrickhulce) [reply .github/workflows/ci.yml:43]: ya we just need those two IIRC, I can get an npm token in the repo 👍

Comment (patrickhulce) [reply .github/workflows/ci.yml:43]: added! let's give it a whirl

Comment (paulirish) [reply .github/workflows/ci.yml:43]: k. just added the env bit into the file.

**Pull request merged**